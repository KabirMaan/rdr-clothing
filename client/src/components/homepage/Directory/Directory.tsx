import React from "react";
import MenuItem from "../MenuItem";
import "./Directory.scss";

interface DirectoryProps {
  sections: {
    title: string;
    imageUrl: string;
    id: number;
    linkUrl: string;
    size: string;
  }[];
  onMenuItemClick: (linkUrl: string) => void;
}

const Directory: React.FC<DirectoryProps> = ({ onMenuItemClick, sections }) => {
  return (
    <div className="directory">
      {sections.map(({ title, imageUrl, id, size, linkUrl }) => {
        return (
          <MenuItem
            key={id}
            imageUrl={imageUrl}
            menuTitle={title}
            size={size}
            onMenuItemClick={() => onMenuItemClick(linkUrl)}
          />
        );
      })}
    </div>
  );
};

export default Directory;
