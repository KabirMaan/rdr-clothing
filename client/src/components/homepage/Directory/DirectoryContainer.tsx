import React from "react";
import Directory from "./Directory";
import { withRouter, RouteComponentProps } from "react-router";
import { createStructuredSelector } from "reselect";
import { selectDirectorySections } from "../../../redux/directory/directorySelectors";
import { connect } from "react-redux";

interface DirectoryContainerProps extends RouteComponentProps {
  sections: {
    title: string;
    imageUrl: string;
    id: number;
    linkUrl: string;
    size: string;
  }[];
}

const DirectoryContainer: React.FC<DirectoryContainerProps> = ({
  history,
  match,
  sections
}) => {
  const onMenuItemClick = (linkUrl: string): void => {
    history.push(`${match.url}${linkUrl}`);
  };

  return <Directory sections={sections} onMenuItemClick={onMenuItemClick} />;
};

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections
});

export default withRouter(connect(mapStateToProps)(DirectoryContainer));
