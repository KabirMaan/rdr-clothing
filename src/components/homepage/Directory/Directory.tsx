import React from "react";
import MenuItem from "../MenuItem";
import "./Directory.scss";
import { withRouter, RouteComponentProps } from "react-router";
import { createStructuredSelector } from "reselect";
import { selectDirectorySections } from "../../../redux/directory/directorySelectors";
import { connect } from "react-redux";

interface DirectoryProps extends RouteComponentProps {
  sections: {
    title: string;
    imageUrl: string;
    id: number;
    linkUrl: string;
    size: string;
  }[];
}

const Directory: React.FC<DirectoryProps> = ({ history, match, sections }) => {
  const onMenuItemClick = (linkUrl: string): void => {
    history.push(`${match.url}${linkUrl}`);
  };

  return (
    <div className="directory">
      {sections.map(({ title, imageUrl, id, size, linkUrl }) => {
        return (
          <MenuItem
            key={id}
            imageUrl={imageUrl}
            menuTitle={title}
            size={size}
            onMenuItemClick={() => onMenuItemClick(linkUrl)}
          />
        );
      })}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections
});

export default withRouter(connect(mapStateToProps)(Directory));
