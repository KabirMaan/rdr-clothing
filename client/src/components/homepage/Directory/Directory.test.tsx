import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Directory from './Directory';


describe("Directory Component", () => {

  let wrapper: ShallowWrapper;
  let mockOnMenuItemClick: jest.Mock<any, any>

  beforeEach(() => {
    mockOnMenuItemClick = jest.fn()

    const mockProps = {
      sections: [],
      onMenuItemClick: mockOnMenuItemClick
    }

    wrapper = shallow(<Directory {...mockProps} />)
  })
  it('should render Directory component', () => {
    expect(wrapper).toMatchSnapshot();
  });
})

