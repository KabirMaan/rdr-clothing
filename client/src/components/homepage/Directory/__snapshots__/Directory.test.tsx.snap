// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Component should render Directory component 1`] = `ShallowWrapper {}`;
